src/dune_rules/melange_rules.ml

```diff
@@ -181,6 +225,7 @@ let add_rules_for_libraries ~dir ~scope ~target_dir ~sctx ~requires_link
       in
       let lib = local_of_lib ~loc:mel.loc lib in
       let info = Lib.Local.info lib in
+      let implements = Lib_info.implements info in
```

Comment:
There's Lib.implements that does resolution for you

Reply:
Using it in https://github.com/ocaml/dune/pull/6495/commits/fb4874199db5ec5d0f3df77485d8ab60c2132874.